docs/source/common/lightning_module.rst

```diff
@@ -484,116 +503,56 @@ There are two ways to call ``test()``:
 
 ----------
 
-Inference (Prediction Loop)
-^^^^^^^^^^^^^^^^^^^^^^^^^^^
-For research, LightningModules are best structured as systems.
-
-.. code-block:: python
-
-    import pytorch_lightning as pl
-    import torch
-    from torch import nn
-
-
-    class Autoencoder(pl.LightningModule):
-        def __init__(self, latent_dim=2):
-            super().__init__()
-            self.encoder = nn.Sequential(nn.Linear(28 * 28, 256), nn.ReLU(), nn.Linear(256, latent_dim))
-            self.decoder = nn.Sequential(nn.Linear(latent_dim, 256), nn.ReLU(), nn.Linear(256, 28 * 28))
-
-        def training_step(self, batch, batch_idx):
-            x, _ = batch
-
-            # encode
-            x = x.view(x.size(0), -1)
-            z = self.encoder(x)
-
-            # decode
-            recons = self.decoder(z)
-
-            # reconstruction
-            reconstruction_loss = nn.functional.mse_loss(recons, x)
-            return reconstruction_loss
-
-        def validation_step(self, batch, batch_idx):
-            x, _ = batch
-            x = x.view(x.size(0), -1)
-            z = self.encoder(x)
-            recons = self.decoder(z)
-            reconstruction_loss = nn.functional.mse_loss(recons, x)
-            self.log("val_reconstruction", reconstruction_loss)
-
-        def predict_step(self, batch, batch_idx, dataloader_idx=0):
-            x, _ = batch
-
-            # encode
-            # for predictions, we could return the embedding or the reconstruction or both based on our need.
-            x = x.view(x.size(0), -1)
-            return self.encoder(x)
+*********
+Inference
+*********
 
-        def configure_optimizers(self):
-            return torch.optim.Adam(self.parameters(), lr=0.0002)
+Prediction Loop
+===============
 
-Which can be trained like this:
+To activate the prediction loop, you may need to override the :meth:`~pytorch_lightning.core.lightning.LightningModule.predict_step` method
+since Lightning under the hood do a forward pass through your LightningModule by default.
 
 .. code-block:: python
 
-    autoencoder = Autoencoder()
-    trainer = pl.Trainer(gpus=1)
-    trainer.fit(autoencoder, train_dataloader, val_dataloader)
-
-This simple model generates examples that look like this (the encoders and decoders are too weak)
-
-.. figure:: https://pl-bolts-doc-images.s3.us-east-2.amazonaws.com/pl_docs/ae_docs.png
-    :width: 300
-
-The methods above are part of the LightningModule interface:
-
-- training_step
-- validation_step
-- test_step
-- predict_step
-- configure_optimizers
+    class LitModel(pl.LightningModule):
+        def predict_step(self, batch, batch_idx):
+            x, y = batch
```

Comment:
Let's show a MCDropout example here and a link to the method usage.

Reply:
done!